NN_chatCmd.js

```diff
@@ -24,156 +167,334 @@ nerthus.chatCmd.fixUrl = function(text)
-    return true;
+    const map_url = ch.t.split(" ").slice(1).join(" ")
+    nerthus.worldEdit.changeMap(map_url, 1)
+
+    return false
 }
 
 nerthus.chatCmd.map["light"] = function(ch)
 {
-    var opacity = ch.t.split(" ")[1];
-    $("#base").css("opacity",opacity);
-    return true;
-}
-
-nerthus.chatCmd.map["addGraf"] = function(ch)
-{  //cmd[0]=x, cmd[1]=y, cmd[2]=url, cmd[3]=tip_text, cmd[4]=isCol
-    var cmd = ch.t.split(" ").slice(1).join(" ").split(",");
-    var x = parseInt(cmd[0]);
-    var y = parseInt(cmd[1]);
-    var _url = cmd[2];
-    var _tip = cmd[3] ? ' tip="<b>'+cmd[3]+'</b>" ctip="t_npc"' : "";
-    var isCol = parseInt(cmd[4]);
-    $('<img id="_ng-' + cmd[0] + '-' + cmd[1] + '" src="'+ _url +'"'+_tip+'>').css("position","absolute").appendTo('#base')
-        .load(function()
-        {  //wyśrodkowanie w osi x i wyrównanie do stóp w osi y
-            var _x = 32 * x + 16 - Math.floor($(this).width() / 2);
-            var _y = 32 * y + 32 - $(this).height();
-            $(this).css({top:"" + _y + "px", left: "" + _x + "px"}).css("z-index", y * 2 + 9);
-        })
-    if(isCol) g.npccol[ x + 256 * y] = true;
-    return true;
+    let opacity = ch.t.split(" ")[1]
+    if(typeof opacity === "undefined")
+        nerthus.worldEdit.changeLight()
+    else
+    {
+        opacity = opacity.replace(",",".")
+        nerthus.worldEdit.changeLight(1 - opacity)
+    }
+
+    return false
+}
+
+nerthus.chatCmd.map["addGraf"] = function (ch)
+{  //cmd[0]=x, cmd[1]=y, cmd[2]=url, cmd[3]=tip_text, cmd[4]=isCol, cmd[5]=map_id
+    const cmd = ch.t.split(" ").slice(1).join(" ").split(",")
+    const x = parseInt(cmd[0])
+    const y = parseInt(cmd[1])
+    const _url = cmd[2]
+    const name = cmd[3]
+    const isCol = parseInt(cmd[4]) > 0
+    const map_id = cmd[5]
+
+    nerthus.worldEdit.addNpc(x, y, _url, name, isCol, map_id)
+
+    return false
 }
 
 nerthus.chatCmd.map["delGraf"] = function(ch)
 {
-    var cmd = ch.t.split(" ")[1].split(",");
-    var x = parseInt(cmd[0]);
-    var y = parseInt(cmd[1]);
-    $('#_ng-' + x + '-' + y).remove();
-    delete g.npccol[x + 256 * y];
-    return true;
+    const cmd = ch.t.split(" ")[1].split(",")
+    const x = parseInt(cmd[0])
+    const y = parseInt(cmd[1])
+    const map_id = cmd[2]
+
+    nerthus.worldEdit.deleteNpc(x, y, map_id)
+
+    return false
+}
+
+nerthus.chatCmd.map["hide"] = function(ch)
+{
+    const cmd = ch.t.split(" ")[1].split(",")
+    const id = parseInt(cmd[0])
+
+    nerthus.worldEdit.hideGameNpc(id)
+
+    return false
 }
 
 nerthus.chatCmd.map["weather"] = function(ch)
 {
     var weather_id = parseInt(ch.t.split(" ")[1])
     nerthus_weather_bard_id = weather_id
-    try
+    nerthus.weather.set_weather(weather_id)
+
+    return false
+}
+
+nerthus.chatCmd.public_map["me"] = function (ch)
+{
+    ch.s = "me"
+    ch.n = ""
+    ch.t = ch.t.replace(/^\*me /, "")
+    return ch
+}
+
+//playing cards
+nerthus.chatCmd.public_map["draw"] = function (ch)
```

Comment:
ta funkcja ma tyle if'ów że gubię się od samego patrzenia

Reply:
to akurat mogę podzielić